instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/CoreRerouteTest.kt

```diff
@@ -281,19 +273,195 @@ class CoreRerouteTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::class.jav
         delay(2000)
         responseModifier.interruptDelay()
 
+        rerouteStateTransitionAssertion.assert()
+    }
+
+    @Test(timeout = 10_000)
+    fun reroute_is_not_cancelled_when_alternatives_change() = sdkTest {
+        val mapboxNavigation = createMapboxNavigation()
+        val mockRoute = RoutesProvider.dc_short_with_alternative(activity)
+        val mockRouteReroute = RoutesProvider.dc_short_with_alternative_reroute(activity)
+        val originLocation = mockRoute.routeWaypoints.first()
+        val originLocationUpdate = mockLocationUpdatesRule.generateLocationUpdate {
+            latitude = originLocation.latitude()
+            longitude = originLocation.longitude()
+        }
+        mockLocationReplayerRule.loopUpdate(originLocationUpdate, times = 1)
+        val offRouteLocation = mockRouteReroute.routeWaypoints.first()
+        val offRouteLocationUpdate = mockLocationUpdatesRule.generateLocationUpdate {
+            latitude = offRouteLocation.latitude()
+            longitude = offRouteLocation.longitude()
+        }
+
+        mockWebServerRule.requestHandlers.addAll(mockRoute.mockRequestHandlers)
+        val requestHandler = MockDirectionsRequestHandler(
+            profile = DirectionsCriteria.PROFILE_DRIVING_TRAFFIC,
+            jsonResponse = mockRoute.routeResponseJson,
+            expectedCoordinates = mockRoute.routeWaypoints,
+            relaxedExpectedCoordinates = false
+        )
+        val rerouteRequestHandler = MockDirectionsRequestHandler(
+            profile = DirectionsCriteria.PROFILE_DRIVING_TRAFFIC,
+            jsonResponse = mockRouteReroute.routeResponseJson,
+            expectedCoordinates = listOf(
+                offRouteLocation,
+                mockRoute.routeWaypoints.last()
+            ),
+            relaxedExpectedCoordinates = true
+        )
+        val responseModifier = DelayedResponseModifier(4000) { it }
```

Comment:
NIT: you can drop the last argument: it will return the original value by default.